contracts/exchange_contract/src/main.sw

```diff
@@ -183,20 +158,17 @@ impl Exchange for Contract {
 
         let total_liquidity = storage.lp_token_supply;
 
-        let eth_amount_key = key_deposits(sender, ETH_ID);
-        let current_eth_amount = get::<u64>(eth_amount_key);
-
-        let token_amount_key = key_deposits(sender, TOKEN_ID);
-        let current_token_amount = get::<u64>(token_amount_key);
+        let current_eth_amount = storage.deposits.get((sender, ~ContractId::from(ETH_ID)));
+        let current_token_amount = storage.deposits.get((sender, ~ContractId::from(TOKEN_ID)));
 
         assert(current_eth_amount > 0);
 
         let mut minted: u64 = 0;
         if total_liquidity > 0 {
             assert(min_liquidity > 0);
 
-            let eth_reserve = get_current_reserve(ETH_ID);
-            let token_reserve = get_current_reserve(TOKEN_ID);
+            let eth_reserve = storage.eth_reserve;
```

Comment:
I was using the ids here because we need to initialize the storage when deploying the contract. How do we know which is the address that represents `storage.eth_reserve`.

Reply:
Ohh I see. I guess that's a workaround until we actually have contract storage? I can revert this part of the change.